content/blogs/newsletter-104.md

```diff
@@ -0,0 +1,33 @@
+---
+title:       "Newsletter 104 - April/May news"
+author:      "GeoB99"
+date:        2024-04-22
+tags:        [ newsletter ]
+---
+
+Greetings from the ReactOS Team!
+In this year we will start with a short summary highlighting the current people's works on the project as well as announcing the selection of new fresh developers.
+In this report we will also talk about the current situation with the symmetric multiprocessing (SMP) support in ReactOS.
+
+## Development team enlargement
+In the early years of this decade the project team base remained for the most part stagnant.
+While the development continued and still continues as usual, the development team hadn't seen any enlargement with potential developers.
+In worst cases some of the existing core developers became inactive for quite a while, of which were maintainers of important core modules of the project.
+
+This left the development workflow in a state of paralysis for some of the module components of ReactOS, as nobody else in the team had the right skills to determine the correctness of the patches provided
+by contributors in order to get them shipped in [master](https://github.com/reactos/reactos) branch.
+The ReactOS Team is currently adopting an approach to circumvent this problem, by enrolling contributors to the development team who are known to have contributed with useful patches for at least a year, have enough coding skills, have collaborated with the project team members and are willing to work on the main field of area they are interested in.
+
+With that said, the ReactOS Team is pleased to announce the upcoming of 4 new developers for the project -- Carl J. Bialorucki (cbialorucki), Doug Lyons (Doug-Lyons), Oleg Dubinskiy (oleg-dubinskiy) and Whindmar Saksit (whindsaks). Their main focus is to maintain the Shell and Audio components respectively, as well as on any module that requires assistance.
+
+# Symmetric Multiprocessing (SMP) support status
+TODO
+
+## User-mode WINE modules syncing
+TODO
+
+## Audio improvements
+TODO
```

Comment:
```suggestion
Several major improvements in audio subsystem have been made by Oleg Dubinskiy recently. In particular, he implemented handling modern audio formats (e.g., Extensible audio, Float audio and others) [6686](https://github.com/reactos/reactos/pull/6686), which support high sample rates, bits per sample sound quality and more than two (stereo) output channels. This allows many 3rd-party programs and games (e.g., AIMP 5.30, QMMP 0.12.17, all Chromium based browsers and GameDevTycoon game) which are using modern formats, to open an audio device and play the sound properly. However, for now the sound is usually working when the audio miniport driver (e. g., Intel AC97, Realtek HD Audio etc.) supports the requested format directly. Otherwise, it won't work, because playing the format which is not supported by driver needs resampling support, which is currently not finished, so is disabled for now in ReactOS by default.
Another improvement he made, is a support for looped wave playback [6761](https://github.com/reactos/reactos/pull/6761). Looped playback functionality allows to play some audio data multiple times (in other words, play it in loop). The audio data to stream is contained in wave headers (each portion of it is in its own `WAVEHDR` structure). In Windows, this is supported by `WHDR_BEGINLOOP` and `WHDR_ENDLOOP` flags together with specifying the amount of loops to play the audio data in `WAVEHDR.dwLoops` member. As a single wave header, same multiple headers can be looped there. Oleg has implemented only looping a single wave header, while looping two or more wave headers is not supported yet. However we have no any testcase for the second case yet, because Oleg did not find any apps those request to loop multiple headers. But, as it was tested with BRD Demo app, looping a single header works properly as it should.
Besides that, he is also working on some another audio improvements (not in the tree yet). For example, `GetWavePosition` API implementation, which retrieves the current wave playback position in time, and is required by many apps to work correctly too. Another improvement some portion of fixes in DirectSound `IKsPropertySet` implementation, which allows to properly detect sound card by VirtualDJ 8.2, same as it's used by Microsoft DirectX Diagnostic Tool as well, and allows to display more information by audio device there. These and more fixes are coming soon! ;)
```
Note this is not the final version, it can be improved still, if needed.

Reply:
Also, as an addendum, maybe it makes sense to insert my last two videos from YouTube and Twitter in the notes, but I don't know how to add them as playable videos, not as just links to the videos.
https://youtu.be/3fNFANlDj3c?si=SdDifN3emmkkXrJC
https://twitter.com/i/status/1783576020907372910

Reply:
Done. Sent them you in PM.